src/shared/utilities/timeoutUtils.ts

```diff
@@ -27,29 +41,31 @@ export class TimeoutError extends Error {
  * @param timeoutLength Length of timeout duration (in ms)
  */
 export class Timeout {
-    private startTime: number
-    private endTime: number
-    private readonly timeoutLength: number
-    private readonly timerPromise: Promise<undefined>
-    private timerTimeout: NodeJS.Timeout
-    private timerResolve!: (value?: Promise<undefined> | undefined) => void
-    private timerReject!: (value?: Error | Promise<Error> | undefined) => void
-    private _completed: boolean = false
+    private _startTime: number
+    private _endTime: number
```

Comment:
Do we have a decisive rule on prefixing with `_` for private things? It's common convention but we're also inconsistent; could lint for it I guess.

Reply:
TECHNICALLY the Javascript thing to do is a `#` for private fields: this actually eliminates the need for an explicit `private` identifier and would actually let us remove the `public` names and have it implicit: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields

THAT BEING SAID: I would rather stay the course with what we have (explicit identifiers and optional `_`), since there's no recourse for `protected` things. Now that this is out in the universe though, I'm open to discussions.

Reply:
Do we even care about `protected` ?

Reply:
Currently? Yes:

![image](https://user-images.githubusercontent.com/29374703/154598525-2cb1aae3-8cc1-4a75-bcbb-b263004be703.png)

Reply:
Good chunk of those are probably from me. I try to use `protected` to signal that this method could be overridden to modify behavior in a (relatively) robust way

I barely use inheritance though so /shrug